### PR TITLE
Add bot attribute and test for common bot strings

### PR DESCRIPTION
**Pull request by triptec**
This might be a better path to add data if it might be a bot.

Copied from a ruby lib called agent orange:
https://github.com/kevinelliott/agent_orange/blob/f06c3d59452cd21f52261b16a6ea1a5b9b8ac0db/lib/agent_orange/device.rb#L35

### CODE REVIEW

File: test/logic.js [hunk 1]
```diff
@@ -222,7 +222,23 @@ test('detects instagram in-app browser', function (t) {
   t.end();
 });
 
+test('detects crawlers', function (t) {
```
Comment (DamonOehlman) [inline test/logic.js]: maybe rename these tests to `detects crawler: <crawler name>`.  Just having a couple of tests is fine too - definitely doesn't need to be exhaustive.

**Comment by triptec** [comment]
Well I work with analytics of user pageviews, clicks and shown ads and the like. Today I'm getting quite a few events that are posted by crawlers that use js and I like to add this data to the events so that it becomes easier to filter those rows in database.

**Comment by DamonOehlman** [comment]
> Well I work with analytics of user pageviews, clicks and shown ads and the like. Today I'm getting quite a few events that are posted by crawlers that use js and I like to add this data to the events so that it becomes easier to filter those rows in database.

Sounds fair to me.  I'll merge this PR and publish a new version to NPM.

**Comment by DamonOehlman** [comment]
Published as `2.3.0`

**Pull request merged**